packages/grafana-ui/src/components/Drawer/Drawer.tsx

```diff
@@ -72,10 +71,14 @@ export const Drawer: FC<Props> = ({
   closeOnMaskClick = false,
   scrollableContent = false,
   title,
+  subtitle,
   width = '40%',
+  expandable = false,
 }) => {
   const theme = useTheme();
   const drawerStyles = getStyles(theme, scrollableContent);
+  const [currentWidth, setCurrentWidth] = useState(width);
+  const isExpanded = currentWidth === '100%';
```

Comment:
Can we make it the other way around? So that `isExpanded` is a state and width is derived from it?